File: mobile/ios/Sources/Domain/UseCases.swift
import Foundation

public struct FetchPolicyUseCase {
    private let repository: PolicyRepository

    public init(repository: PolicyRepository) {
        self.repository = repository
    }

    public func execute() async throws -> Policy {
        try await repository.fetchPolicy()
    }
}

public struct SendTelemetryUseCase {
    private let repository: TelemetryRepository

    public init(repository: TelemetryRepository) {
        self.repository = repository
    }

    public func execute(event: TelemetryEvent) async throws {
        try await repository.send(event: event)
    }
}

File: mobile/android/app/src/main/java/com/panthersecurity/sample/domain/UseCases.kt
package com.panthersecurity.sample.domain

class FetchPolicyUseCase(private val repository: PolicyRepository) {
    suspend fun execute(): Policy = repository.fetchPolicy()
}

class SendTelemetryUseCase(private val repository: TelemetryRepository) {
    suspend fun execute(event: TelemetryEvent) = repository.send(event)
}
